zio-http-cli/src/main/scala/zio/http/endpoint/cli/HttpCliApp.scala

```diff
@@ -0,0 +1,771 @@
+package zio.http.endpoint.cli
+
+import scala.annotation.StaticAnnotation
+import scala.collection.immutable
+import scala.util.Try
+
+import zio._
+import zio.cli._
+import zio.json.ast._
+
+import zio.schema._
+
+import zio.http._
+import zio.http.codec._
+import zio.http.endpoint._
+
+class description(val text: String) extends StaticAnnotation // TODO move this to zio-schema
+
+object HttpCliApp {
+  private[cli] final case class CliRequest(
```

Comment:
I would pull all these nested classes outside of HttpCliApp and make them into separate files, for better organization.